e2e/tests-replica/deploy.bash

```diff
@@ -69,7 +69,7 @@ teardown() {
     dfx_start
     assert_command dfx canister create --all
 
-    assert_command dfx deploy --argument '("World")'
+    assert_command dfx deploy hello_backend --argument '("World")'
```

Comment:
Ideally we'd have some test that demonstrates commands that would have been accepted but now are rejected, both for `dfx deploy` and `dfx canister install`.  Put another way, if a future PR reverted the changes to *.rs files in this PR, no tests would fail.
```suggestion
    assert_command_fail dfx deploy --argument '("World")'
    assert_command dfx deploy hello_backend --argument '("World")'
```